api/v1alpha1/envoyproxy_types.go

```diff
@@ -410,6 +417,18 @@ type EnvoyProxyList struct {
 	Items           []EnvoyProxy `json:"items"`
 }
 
+// IPFamily defines the IP family to use for the Envoy proxy.
+type IPFamily string
+
+const (
+	// IPv4 defines the IPv4 family.
+	IPv4 IPFamily = "IPv4"
+	// IPv6 defines the IPv6 family.
+	IPv6 IPFamily = "IPv6"
+	// DualStack defines the dual-stack family.
+	DualStack IPFamily = "DualStack"
```

Comment:
Nit: Could we add a line to explain how "DualStack" impacts the Envoy proxy fleet here? For example, Envoy proxy will listen on both the IPV4 and IPV6 addresses for incoming traffic.